flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/table/catalog/hive/client/HiveShimV310.java

```diff
@@ -111,4 +112,28 @@ public void makeSpecFromName(Map<String, String> partSpec, Path currPath) {
 		}
 	}
 
+	@Override
+	public Set<String> getNotNullColumns(IMetaStoreClient client, Configuration conf, String dbName, String tableName) {
+		try {
+			Method method = getMetaStoreUtilsClass().getDeclaredMethod("getDefaultCatalog", Configuration.class);
```

Comment:
Is that right to use `getDefaultCatalog`? What about other catalogs?

Reply:
Firstly to clarify, here the "default catalog" refers to the default catalog in HMS, not in Flink. [HIVE-18685](https://issues.apache.org/jira/browse/HIVE-18685) is to introduce catalog to HMS. Since that feature is not yet completed, it's fine to just use `getDefaultCatalog `. After HIVE-18685 is done, we may need to take HMS catalog into consideration when interacting with Hive metadata.

Reply:
Can you add these comments to code?